crates/mdb/src/error.rs

```diff
@@ -0,0 +1,100 @@
+use std::ffi::{c_char, CStr};
+use std::fmt::{Debug, Display, Formatter};
+use std::marker::PhantomData;
+
+unsafe fn pchar_to_string(p_value: *const c_char) -> String {
+    assert!(!p_value.is_null());
+    let value = String::from(CStr::from_ptr(p_value).to_str().unwrap());
+    value
+}
+
+pub trait MdbError<Ptr> {
```

Comment:
What are some benefits of defining a trait for this?

In particular, do we really want `new` and `as_ref` to be available to users of the library?